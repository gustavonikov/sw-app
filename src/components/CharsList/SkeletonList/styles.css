.ant-skeleton-content .ant-skeleton-paragraph {
  margin: 0;
}
