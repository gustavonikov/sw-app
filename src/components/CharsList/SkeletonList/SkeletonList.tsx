import { Card, Skeleton, List } from 'antd'

import './styles.css'

export default function SkeletonList() {
  const skeletonItems = [{},{},{},{},{},{},{},{},{},{}]

  return (
    <List 
      grid={{
        gutter: 24,
        xs: 1,
        sm: 2,
        md: 4,
        lg: 4,
        xl: 5,
        xxl: 5,
      }}
      dataSource={skeletonItems}
      renderItem={() => (
        <List.Item>
          <Card title={'-'}>
            <Skeleton 
              active 
              title={false}
              paragraph={{
                rows: 2,
                width: [100, 100]
              }}
            >
            </Skeleton>
          </Card>
        </List.Item>
      )}
    />
  )
}
