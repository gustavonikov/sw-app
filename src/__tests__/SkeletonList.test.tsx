import { shallow } from 'enzyme'
import SkeletonList from '../components/CharsList/SkeletonList/SkeletonList'

const skeletonList = shallow(<SkeletonList />)

it('renders Skeleton List correctly', () => {
  expect(skeletonList).toMatchSnapshot()
})

