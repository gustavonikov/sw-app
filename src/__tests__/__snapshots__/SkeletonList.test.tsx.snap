// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Skeleton List correctly 1`] = `
<List
  dataSource={
    Array [
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
    ]
  }
  grid={
    Object {
      "gutter": 24,
      "lg": 4,
      "md": 4,
      "sm": 2,
      "xl": 5,
      "xs": 1,
      "xxl": 5,
    }
  }
  renderItem={[Function]}
/>
`;
